pkg/client/elasticsearch/query.go

```diff
@@ -114,17 +116,19 @@ func getMetricForPod(
 	defer res.Body.Close()
 
 	if res.IsError() {
-		var e map[string]interface{}
-		if err := json.NewDecoder(res.Body).Decode(&e); err != nil {
-			return timestampedMetric{}, fmt.Errorf("error parsing the response body: %s", err)
-		} else {
-			// Print the response status and error information.
-			return timestampedMetric{}, fmt.Errorf("[%s] %s: %s",
-				res.Status(),
-				e["error"].(map[string]interface{})["type"],
-				e["error"].(map[string]interface{})["reason"],
-			)
+		bodyBytes, err := io.ReadAll(res.Body)
+		if err != nil {
+			return timestampedMetric{}, errors.Wrap(err, "failed to read body")
+		}
+		var errorResponse ErrorResponse
+		if err := json.Unmarshal(bodyBytes, &errorResponse); err != nil {
+			return timestampedMetric{}, fmt.Errorf("[%s] failed to get metric pod: %s", res.Status(), string(bodyBytes))
```

Comment:
Unmarshall error is lost.
```suggestion
			return timestampedMetric{}, fmt.Errorf("[%s] failed to unmarshal search response '%s' with error %w", res.Status(), string(bodyBytes), err)
```